tensorflow/core/kernels/cwise_op_select.cc

```diff
@@ -324,10 +444,43 @@ struct BatchSelectFunctor<CPUDevice, T> {
   }
 };
 
+template <typename T, int NDIMS>
+struct BCastSelectFunctor<CPUDevice, T, NDIMS> {
```

Comment:
Why is this defined in multiple places? Can you just define this once in the header file and just template it on device type as well?

Reply:
Thanks @rmlarsen, the PR has been updated with two definitions (one for CPU and one for SYCL) consolidated into one.